core/src/com/ericc/the/game/systems/realtime/RenderSystem.java

```diff
@@ -83,14 +81,18 @@ public void update(float deltaTime) {
             }
 
             // Depth-order the entities.
-            visibleEntities.sort((a, b) -> Mappers.renderable.get(b).zOrder - Mappers.renderable.get(a).zOrder);
+            visibleEntities.sort((a, b) -> {
+                if(Mappers.position.get(b).getY() - Mappers.position.get(a).getY() != 0)
```

Comment:
Można użyć takiego comparatora (jest czytelniejsze co tu się dzieje imo)
```
Comparator
                    .comparingInt((Entity e) -> Mappers.position.get(e).getY())
                    .reversed()
                    .thenComparing((Entity e) -> Mappers.renderable.get(e).zOrder)
                    .reversed()
```
Przy czym warto go zapisać sobie gdzieś jako private static

Reply:
i to dajesz jako argument do sort